core/org.eclipse.cdt.core/model/org/eclipse/cdt/core/settings/model/extension/impl/CDefaultConfigurationData.java

```diff
@@ -105,9 +105,17 @@ protected void copySettingsFrom(CConfigurationData base, boolean clone) {
 			return;
 		fDescription = base.getDescription();
 
-		fTargetPlatformData = copyTargetPlatformData(base.getTargetPlatformData(), clone);
+		if (base.getTargetPlatformData() != null) {
```

Comment:
I think it would be cleaner to have the null-check in the `copyTargetPlatformData` and `copyBuildData` functions.

Reply:
Cleaning the code is indeed something that needs to be done. We fully agree on that.
The current code is simply not readable nor understandable for someone with my skill level.
I tried to make the changes very clear, readable, straightforward and safe.
In this case I think code that says: "When there is a targetplatform copy it and assign to x else assign null to x." is way more readable than
Having a copytargetplatform which returns a copy when the source targetplatform is not null and null when the source targetplatform is null. 

But it is core CDT code and as such not my call.